huaweicloud/services/fgs/resource_huaweicloud_fgs_trigger.go

```diff
@@ -0,0 +1,624 @@
+	event["schedule_type"] = d.Get("timer.0.schedule_type").(string)
+	event["user_event"] = d.Get("timer.0.additional_information").(string)
+
+	return event
+}
+
+func buildObsEventName(s *schema.Set) []string {
+	result := make([]string, s.Len())
+	for i, val := range s.List() {
+		switch val.(string) {
+		case obsEventPut, obsEventPost, obsEventCopy, obsEventMultiUpload:
+			// The obs create events are format as 's3:ObjectCreated:{event}'
+			// The events of 'ObjectCreated' are 'Put', 'Post', 'Copy' and 'CompleteMultipartUpload'.
+			result[i] = fmt.Sprintf("s3:%s:%s", obsEventCreated, val.(string))
+		case obsEventDelete, obsEventDeleteWithoutMarker:
+			// The obs remove events are format as 's3:ObjectRemoved:{event}'
+			// The events of 'ObjectRemoved' are 'Delete' and 'DeleteMarkerCreated'.
+			result[i] = fmt.Sprintf("s3:%s:%s", obsEventRemoved, val.(string))
+		default:
+			// The obs events are format as 's3:ObjectCreated:*' or 's3:ObjectRemoved:*'
+			result[i] = fmt.Sprintf("s3:%s:*", val.(string))
+		}
+	}
+	return result
+}
+
+func buildObsEventData(d *schema.ResourceData) map[string]interface{} {
+	event := make(map[string]interface{})
+
+	obsEvents := d.Get("obs.0.events").(*schema.Set)
+	event["bucket"] = d.Get("obs.0.bucket_name").(string)
+	event["events"] = buildObsEventName(obsEvents)
+	event["name"] = d.Get("obs.0.event_notification_name").(string)
+	if prefix, ok := d.GetOk("obs.0.prefix"); ok {
+		event["prefix"] = prefix.(string)
+	}
+	if suffix, ok := d.GetOk("obs.0.suffix"); ok {
+		event["suffix"] = suffix.(string)
+	}
+
+	return event
+}
+
+func buildSmnEventData(d *schema.ResourceData) map[string]interface{} {
+	return map[string]interface{}{
+		"topic_urn": d.Get("smn.0.topic_urn").(string),
+	}
+}
+
+func buildDisEventData(d *schema.ResourceData) map[string]interface{} {
+	return map[string]interface{}{
+		"stream_name":        d.Get("dis.0.stream_name").(string),
+		"sharditerator_type": d.Get("dis.0.starting_position").(string),
+		"max_fetch_bytes":    d.Get("dis.0.max_fetch_bytes").(int),
+		"polling_interval":   d.Get("dis.0.pull_period").(int),
+		"polling_unit":       "ms",
+		"is_serial":          strconv.FormatBool(d.Get("dis.0.serial_enable").(bool)),
+	}
+}
+
+func buildKafkaEventData(d *schema.ResourceData) map[string]interface{} {
+	return map[string]interface{}{
+		"instance_id": d.Get("kafka.0.instance_id").(string),
+		"batch_size":  d.Get("kafka.0.batch_size").(int),
+		"topic_ids":   utils.ExpandToStringListBySet(d.Get("kafka.0.topic_ids").(*schema.Set)),
+	}
+}
+
+func buildFunctionGraphTriggerParameters(d *schema.ResourceData, config *config.Config) (trigger.CreateOpts, error) {
+	triggerType := d.Get("type").(string)
+
+	opts := trigger.CreateOpts{
+		TriggerTypeCode: triggerType,
+		TriggerStatus:   d.Get("status").(string),
+	}
+
+	switch triggerType {
+	case timingTrigger:
+		opts.EventData = buildTimingEventData(d)
+		opts.EventTypeCode = "MessageCreated"
```

Comment:
suggest to set **EventTypeCode** when initializing the CreateOpts.

Reply:
The initialization `EventTypeCode` has been updated.